### PR TITLE
Make analyzer skip files with non yaml or json extension

### PR DESCRIPTION
**Pull request by jasonwzm**
Please provide a description for what this PR is for.
fix https://github.com/istio/istio/issues/21712
We are already handling parsing error. This fix just skips the files with non yaml or json extension and prints out a message.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/galley/analyze_test.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ const (
 	serviceRoleBindingFile = "testdata/servicerolebinding.yaml"
 	serviceRoleFile        = "testdata/servicerole.yaml"
 	invalidFile            = "testdata/invalid.yaml"
+	invalidExtensionFile   = "testdata/invalid.md"
```
Comment (liamawhite) [inline tests/integration/galley/analyze_test.go]: Could you add test cases for .yml and .json as well?

**Comment by esnible** [comment]
I approved but added `hold` so you can add the test case requested by Liam.

File: tests/integration/galley/analyze_test.go (hunk 1 above)
Comment (jasonwzm) [reply tests/integration/galley/analyze_test.go]: Done. Added a test for json. Yaml is tested in existing testing cases.

**Pull request merged**